python/lib/dependabot/python/file_updater/poetry_file_updater.rb

```diff
@@ -259,8 +266,15 @@ def pyproject_hash_for(pyproject_content)
         end
 
         def declaration_regex(dep)
-          escaped_name = Regexp.escape(dep.name).gsub("\\-", "[-_.]")
-          /(?:^\s*|["'])#{escaped_name}["']?\s*=.*$/i
+          /(?:^\s*|["'])#{escape(dep)}["']?\s*=.*$/i
+        end
+
+        def table_declaration_regex(dep)
+          /tool\.poetry\.[^\n]+\.#{escape(dep)}\]\n.*?\s*version\s* =.*?\n/m
```

Comment:
I think this regex might fail if `version` is not the first key listed for the table dependency? This was an issue that came up for Terraform

Reply:
I think it should work, because in multiline regexp's `.*` should match anything _including_ new lines. I _think_ I also verified this manually but I will double check and add a new spec for that just in case.

Reply:
> I see /m enables multiline mode, so maybe the \n.* will be able to capture until the version key is found.

Exactly! I also used non-greedy operator so that we don't accidentally match the `version =` field of a different table dependency and replace the wrong thing.

> I think it is still a good idea still to include a test where version is not the first key in the table dependency. That way we can know if that breaks if this regex changes

Agreed, I will add more tests for both of the above cases 👍.

Reply:
New tests added!